lib/Dialect/FIRRTL/Transforms/CheckCombLoops.cpp

```diff
@@ -398,27 +398,52 @@ class DiscoverLoops {
   }
 
   void reportLoopFound(Value childVal, VisitingSet visiting) {
-    auto getName = [&](FieldRef node) {
-      if (isa<SubfieldOp, SubindexOp, SubaccessOp>(node.getDefiningOp())) {
-        assert(!valRefersTo[node.getValue()].empty());
-        return getFieldName(*valRefersTo[node.getValue()].begin()).first;
+    auto getName = [&](Value v) {
+      if (isa_and_nonnull<SubfieldOp, SubindexOp, SubaccessOp>(
+              v.getDefiningOp())) {
+        assert(!valRefersTo[v].empty());
+        // TODO: Indicate to user if "alias set" if > 1, and/or pick
```

Comment:
We don't have enough information here, but this can be deterministic if we do another traversal to find the actual value that's connected, and just print that. We want the loop detection to be fast, but the error reporting can be slower. So, given the source and sink of the cycle, find the actual path between them and report the loop.

Reply:
Plus, we need to handle `SubAccessOp` in `getFieldName` .